packages/core/src/components/fields/ObjectField.js

```diff
@@ -8,24 +8,31 @@ import {
   getDefaultRegistry,
   canExpand,
   ADDITIONAL_PROPERTY_FLAG,
+  getUiOptions,
 } from "../../utils";
 
 function DefaultObjectFieldTemplate(props) {
   const { TitleField, DescriptionField } = props;
+  const uiOptions = getUiOptions(props.uiSchema);
+  let { label: displayLabel = true } = uiOptions;
+
+  const title = uiOptions.title || props.title;
```

Comment:
I noticed that we're changing the order of precedence here (uioptions.title first and then props.title). This is OK (and good so that we're consistent), but I think we should probably label this a breaking change? What do you think?

if so, can you make a "4.x upgrade guide.md" in the docs and describe this change to users?

Reply:
@epicfaace I'm not sure this is the case.
If  you look [below](https://github.com/rjsf-team/react-jsonschema-form/blob/93c278b11111eaa031eb5ccd3e8045c84ce7e344/packages/core/src/components/fields/ObjectField.js#L231) on this file, you can see that `props.title` is actually `uiSchema["ui:title"] || title`
This seems to be a little convoluted I agree, we are looking at the uiSchema both in the field definition and in the template. But in any case, the behaviour should be consistent with the current version

Reply:
Oh, then that's great! What do you think @newt10 ? I think we can remove the 4.x upgrade guide changes then?

Reply:
@bejito thats a good catch, this fix is not inverting the order. @epicfaace  I confirmed it on the playground with before and after the fix

#Before - ui:title is taking priority
![Screen Shot 2021-09-06 at 11 55 04 AM](https://user-images.githubusercontent.com/42853152/132254332-b5f09e46-3c31-40d6-aa9c-e9a8aaed421c.png)

#After - ui:options takes priority if provided
![Screen Shot 2021-09-06 at 11 58 17 AM](https://user-images.githubusercontent.com/42853152/132254341-dbd3b9ef-ce9c-4845-9b4a-541a07ffc4ec.png)